### ISSUE TITLE
unable to run read.maf

### ISSUE THREAD

**Issue by skr2345** [PoisonAlien/maftools#128]
since update of R to 3.4.3 and Bioconductor to 3.6 (Mac OS X) I am not able to run read.maf anymore. The following message appears:

+                      > maf_all <-  read.maf(maf = all, removeSilent = FALSE, useAll = TRUE)

+ > **Fehler (=mistake) in read.maf(maf = all, removeSilent = FALSE, useAll = TRUE) : 
  unbenutztes (=unused) Argument (removeSilent = FALSE)**

As soon I run this script without any argument, I have the following warning after "Summarizing" step:

+                      > maf_all <-  read.maf(maf = all)
+ > Summarizing.. 
+ > Fehler (=mistake) in dcast.data.table(data = vc, formula = Tumor_Sample_Barcode ~  : 
  Can not cast an empty data.table

**Comment by PoisonAlien** [comment]
Hello,

Can you post your sessionInfo ?

**Comment by skr2345** [comment]
here is the sessionInfo with loaded maftools and data.table:

----------------------
R version 3.4.3 (2017-11-30)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] de_DE.UTF-8/de_DE.UTF-8/de_DE.UTF-8/C/de_DE.UTF-8/de_DE.UTF-8

attached base packages:
character(0)

other attached packages:
[1] maftools_1.4.25

loaded via a namespace (and not attached):
 [1] httr_1.3.1                 Biobase_2.38.0            
 [3] RMySQL_0.10.13             bit64_0.9-7               
 [5] splines_3.4.3              foreach_1.4.4             
 [7] assertthat_0.2.0           stats4_3.4.3              
 [9] grDevices_3.4.3            blob_1.1.0                
[11] BSgenome_1.46.0            GenomeInfoDbData_1.0.0    
[13] Rsamtools_1.30.0           slam_0.1-42               
[15] progress_1.1.2             ggrepel_0.7.0             
[17] pillar_1.2.0               RSQLite_2.0               
[19] lattice_0.20-35            base_3.4.3                
[21] digest_0.6.15              GenomicRanges_1.30.2      
[23] RColorBrewer_1.1-2         XVector_0.18.0            
[25] colorspace_1.3-2           cowplot_0.9.2             
[27] Matrix_1.2-12              plyr_1.8.4                
[29] XML_3.98-1.10              GetoptLong_0.1.6          
[31] biomaRt_2.34.2             zlibbioc_1.24.0           
[33] xtable_1.8-2               scales_0.5.0              
[35] BiocParallel_1.12.0        tibble_1.4.2              
[37] pkgmaker_0.22              datasets_3.4.3            
[39] IRanges_2.12.0             ggplot2_2.2.1             
[41] SummarizedExperiment_1.8.1 GenomicFeatures_1.30.3    
[43] BiocGenerics_0.24.0        lazyeval_0.2.1            
[45] mclust_5.4                 survival_2.41-3           
[47] magrittr_1.5               memoise_1.1.0             
[49] methods_3.4.3              doParallel_1.0.11         
[51] changepoint_2.2.2          NMF_0.20.6                
[53] utils_3.4.3                tools_3.4.3               
[55] registry_0.5               data.table_1.10.4-3       
[57] prettyunits_1.0.2          GlobalOptions_0.0.12      
[59] matrixStats_0.53.1         gridBase_0.4-7            
[61] ComplexHeatmap_1.17.1      stringr_1.3.0             
[63] S4Vectors_0.16.0           munsell_0.4.3             
[65] cluster_2.0.6              rngtools_1.2.4            
[67] DelayedArray_0.4.1         AnnotationDbi_1.40.0      
[69] stats_3.4.3                Biostrings_2.46.0         
[71] compiler_3.4.3             GenomeInfoDb_1.14.0       
[73] rlang_0.2.0                grid_3.4.3                
[75] RCurl_1.95-4.10            iterators_1.0.9           
[77] VariantAnnotation_1.24.5   rjson_0.2.15              
[79] graphics_3.4.3             circlize_0.4.3            
[81] bitops_1.0-6               gtable_0.2.0              
[83] codetools_0.2-15           DBI_0.7                   
[85] reshape2_1.4.3             R6_2.2.2                  
[87] gridExtra_2.3              zoo_1.8-1                 
[89] GenomicAlignments_1.14.1   rtracklayer_1.38.3        
[91] bit_1.1-12                 shape_1.4.4               
[93] stringi_1.1.6              parallel_3.4.3            
[95] Rcpp_0.12.15               wordcloud_2.5

**Comment by PoisonAlien** [comment]
Hi, 

In removeSilent argument has been deprecated so no need to use it. For your second error, how many samples do you have in your maf ? Or your maf has no non-syn mutations. Can you check ?

**Comment by skr2345** [comment]
yes, I have synonymous mutations. Here is the summary of the same data under Win7 with previous version of R and Bioconductor:

NOTE: Removed 134 duplicated variants
NOTE: Non MAF specific values in Variant_Classification column:
[1] "SNP" "INS" "DEL"
NOTE: Non MAF specific values in Variant_Type column:
[1] "UTR"    "CODING" "INTRON" "IGR"    "RNA"  
Mutation_Status not found. Assuming all variants are Somatic and validated.
Silent variants are being kept!
Creating oncomatrix (this might take a while)..
Sorting..
Summarizing..
           ID summary       Mean Median
1: NCBI_Build      NA         NA     NA
2:     Center      NA         NA     NA
3:    Samples      88         NA     NA
4:     nGenes   15339         NA     NA
5:        DEL    1264   14.36364    9.0
6:        INS    3072   34.90909   23.5
7:        SNP  122812 1395.59091  757.5
8:      total  127148 1444.86364  813.5
Frequently mutated genes..
       Hugo_Symbol DEL INS SNP total MutatedSamples
   1:         TTN   0   7 482   489             76
    2:       MUC16   0   2 302   304             62
    3:     CACNA1E   0   1 101   102             48
    4:       OBSCN   0   0 118   118             45
    5:       SYNE1   1   0  87    88             45
   ---                                            
15335:      ZNF787   0   0   1     1              1
15336:       ZNRD1   0   0   1     1              1
15337:      ZRANB2   0   0   1     1              1
15338:        ZW10   0   0   1     1              1
15339:      ZWILCH   0   0   1     1              1
NOTE: Possible FLAGS among top ten genes:
[1] "TTN"   "MUC16" "OBSCN" "SYNE1" "NEB" 
Done !
---------------------------

But if I run it without arguments and under new R and Bioconductor, the same data produce totally different summary:

NOTE: Removed 134 duplicated variants
NOTE: Non MAF specific values in Variant_Classification column:
[1] "SNP" "INS" "DEL"
NOTE: Non MAF specific values in Variant_Type column:
[1] "UTR"    "CODING" "INTRON" "IGR"    "RNA"   
silent variants: 127148
        ID      N
1: Samples     88
2:     DEL   1264
3:     INS   3072
4:     SNP 122812
Summarizing..
Fehler (=mistake) in dcast.data.table(data = vc, formula = Tumor_Sample_Barcode ~  : 
  Can not cast an empty data.table

**Comment by PoisonAlien** [comment]
Hi,

I think there is an issue with your MAF file. These ("SNP" "INS" "DEL") should be under Variant_Classification and these ("UTR" "CODING" "INTRON" "IGR" "RNA") should be under Variant_classification. Please see MAF file specification for details and correct your maf file.

**Comment by skr2345** [comment]
Thank you for your support! I did not realize that I started to use the newer version of maftools mistakenly thinking that the problems appeared because of update of Bioconductor.
Here is the solution of my issue: one should provide manual list of variant classifications to be considered as non-synonymous using parameter "vc_nonSyn". So in my case this is the list of c('SNP','DEL','INS').
Again, thank you very much for your quick response!
